diff --git a/hw/vfio/common.c b/hw/vfio/common.c
index a9d99919d2..e0ba567dd7 100644
--- a/hw/vfio/common.c
+++ b/hw/vfio/common.c
@@ -667,6 +667,7 @@ static void vfio_iommu_map_notify(IOMMUNotifier *n, IOMMUTLBEntry *iotlb)
     }
 
     rcu_read_lock();
+    goto out;
 
     if ((iotlb->perm & IOMMU_RW) != IOMMU_NONE) {
         bool read_only;
@@ -1073,6 +1074,8 @@ static void vfio_listener_region_add(MemoryListener *listener,
         }
     }
 
+    return;
+
     ret = vfio_dma_map(container, section->mr, iova, int128_get64(llsize),
                        vaddr, section->readonly);
     if (ret) {
@@ -1195,7 +1198,8 @@ static void vfio_listener_region_del(MemoryListener *listener,
         try_unmap = false;
     }
 
-    if (try_unmap) {
+    /*if (try_unmap) {*/
+    if (false) {
         if (int128_eq(llsize, int128_2_64())) {
             /* The unmap ioctl doesn't accept a full 64-bit span. */
             llsize = int128_rshift(llsize, 1);
diff --git a/hw/vfio/pci.c b/hw/vfio/pci.c
index 9b46ec633a..3429495841 100644
--- a/hw/vfio/pci.c
+++ b/hw/vfio/pci.c
@@ -2259,7 +2259,7 @@ static int vfio_pci_hot_reset(VFIOPCIDevice *vdev, bool single)
     info = g_malloc0(sizeof(*info));
     info->argsz = sizeof(*info);
 
-    ret = ioctl(vdev->vbasedev.fd, VFIO_DEVICE_GET_PCI_HOT_RESET_INFO, info);
+    ret = ioctl(vdev->vbasedev.fd, VFIO_DEVICE_GET_PCI_HOT_RESET_INFO, info); // MARKER i feel like we should take a look at HOT_RESET for vmux
     if (ret && errno != ENOSPC) {
         ret = -errno;
         if (!vdev->has_pm_reset) {
@@ -2961,7 +2961,7 @@ static int vfio_interrupt_setup(VFIOPCIDevice *vdev, Error **errp)
     return 0;
 }
 
-static void vfio_realize(PCIDevice *pdev, Error **errp)
+static void vfio_realize(PCIDevice *pdev, Error **errp) // MARKER entry point?
 {
     VFIOPCIDevice *vdev = VFIO_PCI_BASE(pdev);
     VFIODevice *vbasedev_iter;
