diff --git a/hw/vfio/common.c b/hw/vfio/common.c
index a9d99919d2..acba6b10ef 100644
--- a/hw/vfio/common.c
+++ b/hw/vfio/common.c
@@ -2859,11 +2859,12 @@ static int vfio_io_dma_map(VFIOContainer *container, MemoryRegion *mr,
      * again.  This shouldn't be necessary, but we sometimes see it in
      * the VGA ROM space.
      */
+        printf("MAP_DMA:\nflags: 0x%x\nvaddr: 0x%llx\niova: 0x%llx\nsize: 0x%llx\n",map->flags,map->vaddr,map->iova,map->size);
     if (ioctl(container->fd, VFIO_IOMMU_MAP_DMA, map) == 0 ||
         (errno == EBUSY &&
          vfio_dma_unmap(container, map->iova, map->size, NULL) == 0 &&
          ioctl(container->fd, VFIO_IOMMU_MAP_DMA, map) == 0)) {
-        return 0;
+	return 0;
     }
     return -errno;
 }
